src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/get_interior_vehicle_data_request.cc

```diff
@@ -0,0 +1,295 @@
+                       hmi_capabilities,
+                       policy_handle,
+                       resource_allocation_manager)
+    , excessive_subscription_occured_(false) {}
+
+bool CheckIfModuleTypeExistInCapabilities(
+    const smart_objects::SmartObject& rc_capabilities,
+    const std::string& module_type) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (enums_value::kRadio == module_type &&
+      !rc_capabilities.keyExists(strings::kradioControlCapabilities)) {
+    LOG4CXX_DEBUG(logger_, " Radio control capabilities not present");
+    return false;
+  }
+  if (enums_value::kClimate == module_type &&
+      !rc_capabilities.keyExists(strings::kclimateControlCapabilities)) {
+    LOG4CXX_DEBUG(logger_, " Climate control capabilities not present");
+    return false;
+  }
+
+  return true;
+}
+
+void GetInteriorVehicleDataRequest::Execute() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  const smart_objects::SmartObject* rc_capabilities =
+      hmi_capabilities_.rc_capability();
+
+  const std::string module_type = ModuleType();
+  if (rc_capabilities &&
+      !CheckIfModuleTypeExistInCapabilities(*rc_capabilities, module_type)) {
+    LOG4CXX_WARN(logger_, "Accessing not supported module data");
+    SetResourceState(ModuleType(), ResourceState::FREE);
+    SendResponse(false,
+                 mobile_apis::Result::UNSUPPORTED_RESOURCE,
+                 "Accessing not supported module data");
+    return;
+  }
+
+  app_mngr::ApplicationSharedPtr app =
+      application_manager_.application(connection_key());
+
+  if (HasRequestExcessiveSubscription()) {
+    excessive_subscription_occured_ = true;
+    is_subscribed =
+        (*message_)[app_mngr::strings::msg_params][message_params::kSubscribe]
+            .asBool();
+    RemoveExcessiveSubscription();
+  }
+
+  (*message_)[app_mngr::strings::msg_params][app_mngr::strings::app_id] =
+      app->app_id();
+
+  SendHMIRequest(hmi_apis::FunctionID::RC_GetInteriorVehicleData,
+                 &(*message_)[app_mngr::strings::msg_params],
+                 true);
+}
+
+void GetInteriorVehicleDataRequest::on_event(
+    const app_mngr::event_engine::Event& event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  RCCommandRequest::on_event(event);
+
+  if (hmi_apis::FunctionID::RC_GetInteriorVehicleData != event.id()) {
+    return;
+  }
+
+  smart_objects::SmartObject& hmi_response =
+      const_cast<smart_objects::SmartObject&>(event.smart_object());
+
+  mobile_apis::Result::eType result_code =
+      GetMobileResultCode(static_cast<hmi_apis::Common_Result::eType>(
+          hmi_response[app_mngr::strings::params][app_mngr::hmi_response::code]
+              .asUInt()));
+
+  bool result =
+      helpers::Compare<mobile_apis::Result::eType, helpers::EQ, helpers::ONE>(
+          result_code,
+          mobile_apis::Result::SUCCESS,
+          mobile_apis::Result::WARNINGS);
+
+  if (mobile_apis::Result::READ_ONLY == result_code) {
+    result = false;
+    result_code = mobile_apis::Result::GENERIC_ERROR;
```

Comment:
why translate READ_ONLY to GENERIC_ERROR?

Reply:
@yang1070 because for GetInteriorVehicleData READ_ONLY - this is **not valid** response from HMI: https://github.com/smartdevicelink/sdl_requirements/blob/master/detailed_docs/RC/current_module_status_data.md
*In case SDL gets (any-RPC-except-of-SetInteriorVehicleData)-response (resultCode: READ_ONLY) from HMI,
SDL must send (any-RPC-except-of-SetInteriorVehicleData)-response (resultCode: GENERIC_ERROR, success: false) to the related mobile app.*

Reply:
@LitvinenkoIra thank you